Source/Core/Core/PowerPC/Interpreter/Interpreter_SystemRegisters.cpp

```diff
@@ -487,50 +487,66 @@ void Interpreter::mtspr(UGeckoInstruction inst)
 
 void Interpreter::crand(UGeckoInstruction inst)
 {
-  PowerPC::ppcState.cr.SetBit(inst.CRBD, PowerPC::ppcState.cr.GetBit(inst.CRBA) &
-                                             PowerPC::ppcState.cr.GetBit(inst.CRBB));
+  const u32 a = PowerPC::ppcState.cr.GetBit(inst.CRBA);
+  const u32 b = PowerPC::ppcState.cr.GetBit(inst.CRBB);
+
+  PowerPC::ppcState.cr.SetBit(inst.CRBD, a & b);
```

Comment:
Is there a reason why `GetBit` and `SetBit` are using `u32` instead of `bool`? It may be cleaner to change the definitions of those functions than to start using `u32` here.

Reply:
Using bool here can result in false-positives (e.g. `bool & bool` and `bool | bool` are moderately common logic bugs, so compilers and analysis tools will usually warn about these or have facilities that find them).

FWIW, these functions were fine as they were conversion-wise, I just hoisted the accessors out of the `SetBit` call to make them a little nicer to read, given the accessors themselves are quite long.

Reply:
Alright, let's keep this as it is in this PR then.